### PR TITLE
Vocabulary improvement Phase 1

### PR DESCRIPTION
**Pull request by iherman**
This is the first of several steps to clean up the Credentials Vocabulary. The main changes are:

- Anchors have been added to all term definition to allow for an external reference. That is the only change on the spec text itself, no content has been changed. Note that this change takes care of issue #1080
- For all properties, and also for some of the classes, a `defined by` tag has been added in the (YAML) vocabulary definition. This is translated into the `rdfs:isDefinedBy` statement in the formal vocabularies, and a textual translation thereof in the HTML text. Any other text in the property definitions had been removed. Bottom line: the official term definition is the in VCWG specifications, the vocabulary just "refers" to that. Some additional comments:
  - There is one term in the vocabulary, namely `serviceEndpoint` that has ***no*** formal definition. The vocabulary has a comment on this, and this must be solved asap
  - All terms are actually defined in the VCDM spec. The only exception is the `JsonSchema2023` class is defined by the [schema spec](https://www.w3.org/TR/vc-json-schema/#jsonschema2023). 
  - The deprecated terms refer back to the v1.1 of the VCDM specification.
  - Some classes have no direct reference in the spec, their roles are, sort of, "scaffolding" in the vocabulary. This is perfectly fine, they are intermediary classes and carry no other role.
  - The definition of deprecated terms is kept to the bare minimum. In particular, no range or domain statement on other elements on the vocabulary are kept. After all, these are mostly placeholder items, to avoid 404-s on their relevant identifiers.
  - This part of the PR takes care of issue #1061

### For reviewers

The PR does not include the visible files, i.e., the vocabulary in HTML/JSON-LD/Turtle. This is because the generation of these files happen automatically by a GitHub action. To make the reviews possible:

- The files have been generated separately, and can be reached from [the relevant index file](https://w3c.github.io/yml2vocab/previews/vcdm/) in the tool repository for review.
- The textual changes are to be made on the `vocab/credentials/v2/vocabulary.yml` and, for the text framework in the generated HTML, on the `vocab/credentials/v2/template.html` files. (These are the vocabulary generation input files.)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/pull/1209.html" title="Last updated on Jul 27, 2023, 10:00 PM UTC (ee9a7ff)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/1209/7cac20d...ee9a7ff.html" title="Last updated on Jul 27, 2023, 10:00 PM UTC (ee9a7ff)">Diff</a>

### CODE REVIEW

File: vocab/credentials/v2/vocabulary.yml
Comment (OR13) [inline vocab/credentials/v2/vocabulary.yml:6]: Prefer removing rather than commenting.

File: vocab/credentials/v2/vocabulary.yml
Comment (OR13) [inline vocab/credentials/v2/vocabulary.yml:132]: This seems like DID Core Terminology.

File: vocab/credentials/v2/vocabulary.yml
Comment (OR13) [inline vocab/credentials/v2/vocabulary.yml:161]: I don't understand what this is saying

**Review by OR13 — APPROVED** [review]

File: vocab/credentials/v2/vocabulary.yml
Comment (iherman) [reply vocab/credentials/v2/vocabulary.yml:6]: I did not see any resolution on that so far, so I preferred to put it into a comment to be sure there is a consensus around this. Personally, I am in favor of removing it...

File: vocab/credentials/v2/vocabulary.yml
Comment (iherman) [reply vocab/credentials/v2/vocabulary.yml:132]: Maybe... but how did this land in this vocabulary?

File: vocab/credentials/v2/vocabulary.yml
Comment (iherman) [reply vocab/credentials/v2/vocabulary.yml:161]: Which part? The main sentence or the parenthetical one?

The value of the property is _not_ a verifiable credential. It is a separate (rdf) graph that contains a (single) verifiable credential. This is what the JSON-LD `"@container": "@graph"` statement in the `@context` file leads to in terms of RDF.

File: vocab/credentials/v2/vocabulary.yml
Comment (TallTed) [inline vocab/credentials/v2/vocabulary.yml:115]:
```suggestion
    comment: The value of this property must be a URL uniquely identifying the issuer.
```

File: vocab/credentials/v2/vocabulary.yml
Comment (TallTed) [inline vocab/credentials/v2/vocabulary.yml:135]:
```suggestion
    comment: The value of this property must be a URL to the service endpoint associated with the subject. <b><em>There is no URL yet in a VCWG document to <em>define</em> this term.</em></b> 
```

File: vocab/credentials/v2/vocabulary.yml
Comment (TallTed) [inline vocab/credentials/v2/vocabulary.yml:161]:
```suggestion
    comment: The value of this property identifies a <a href="#VerifiableCredentialGraph">Verifiable credential graph</a>. (Informally, it indirectly identifies a <a href="#VerifiableCredential">Verifiable credential</a> contained in a separate graph.)
```

**Review by TallTed — CHANGES_REQUESTED** [review]
I further wonder about the several terms that are left without `comment` descriptions.

File: vocab/credentials/v2/vocabulary.yml
Comment (OR13) [reply vocab/credentials/v2/vocabulary.yml:40]: Text suggestion is good... Refresh service seems very related to protocols... Where will they be defined? Do refresh services work with OIDC?

Probably questions for the core data model.

File: vocab/credentials/v2/vocabulary.yml
Comment (OR13) [reply vocab/credentials/v2/vocabulary.yml:77]: Perhaps we can snag updates from the presentation related PRs for this section.

**Review by msporny — APPROVED** [review]
This is an improvement over what we have today. Approving given that @TallTed and @OR13's comments are addressed.

**Review by dlongley — APPROVED** [review]

**Comment by iherman** [comment]
> I further wonder about the several terms that are left without `comment` descriptions.

The guideline is that comments should not appear if there is a "defined by" field: go to the official spec for the definition.

This is not an absolute rule, there may be cases when an extra local explanation is necessary, like in the case of the usage of RDF Graphs. Or when the class is really some sort of local, RDF term that is more or less necessary to provide a consistent structure to the vocabulary (e.g., common supertypes for classes that are used in practice as objects of properties). But those are (and should be) rare.

File: vocab/credentials/v2/vocabulary.yml
Comment (iherman) [reply vocab/credentials/v2/vocabulary.yml:77]: Do you propose to keep it for now and then handle it in a separate issue/PR? That is fine with me. I do not think the discussion should block the merge of this PR...

File: vocab/credentials/v2/vocabulary.yml
Comment (iherman) [reply vocab/credentials/v2/vocabulary.yml:40]: > Probably questions for the core data model.

indeed, and I would prefer to handle that in a separate issue/PR. The vocabulary related issues/PRs should be following these types of discussions and not be the place to have the discussions themselves...

**Comment by TallTed** [comment]
> The guideline is that comments should not appear if there is a "defined by" field: go to the official spec for the definition.

OK. That was not clear from reading the PR. It might be worth adding such a statement to the description of the Credentials Vocabulary itself; inline and near the start, if possible.

**Review by pchampin — CHANGES_REQUESTED** [review]
I noticed several issues with the generated Turtle and JSON-LD files.
One of them is due to an error in the Yaml file (for which I requested a change).
Others are due to issues in yml2vocab itself, for which I made two PRs w3c/yml2vocab#13 and w3c/yml2vocab#14

**Review by decentralgabe — APPROVED** [review]

**Comment by iherman** [comment]
> I noticed several issues with the generated Turtle and JSON-LD files. One of them is due to an error in the Yaml file (for which I requested a change). Others are due to issues in yml2vocab itself, for which I made two PRs [w3c/yml2vocab#13](https://github.com/w3c/yml2vocab/pull/13) and [w3c/yml2vocab#14](https://github.com/w3c/yml2vocab/pull/14)

For the records: the ymlvocab errors have been taken care of, and a new version of the npm library published.

File: vocab/credentials/v2/vocabulary.yml
Comment (TallTed) [reply vocab/credentials/v2/vocabulary.yml:77]: @iherman -- A fine plan. Might be worth memorializing whatever we plan to do (or review) later, after merging this PR, in new issue(s).

File: vocab/credentials/v2/template.html
Comment (TallTed) [inline vocab/credentials/v2/template.html:142]:
```suggestion
        In some cases, a local explanation is necessary to complement, or to replace, the definition found in an external specification. For instance, this is so when the term is needed to provide a consistent structure to the RDFS vocabulary, such as when the term defines a common supertype for class instances that are used as objects of specific properties, or when <a href="https://www.w3.org/TR/rdf12-concepts/#section-rdf-graph">RDF Graphs</a> are involved. For such cases, the extra definition is included in the current document (and the `rdfs:comment` property is used to include them in the RDFS representations). 
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Minor tweaks.

File: vocab/credentials/v2/vocabulary.yml
Comment (iherman) [reply vocab/credentials/v2/vocabulary.yml:77]: @TallTed could I rely on you raising a new issue when the time comes regarding this specific question? This is your question after all... I will take care of the other issues (essentially, missing term definitions in the spec).

File: vocab/credentials/v2/template.html
Comment (iherman) [reply vocab/credentials/v2/template.html:142]: Thanks @TallTed (for both).

**Comment by iherman** [comment]
I believe all outstanding questions/suggestions have been addressed. From my point of view, this is ready to be merged...

File: vocab/credentials/v2/template.html
Comment (TallTed) [inline vocab/credentials/v2/template.html:174]:
```suggestion
        Implementers may use these properties, but should expect them and/or their meanings to change during the process of
        normatively specifying them.
```

File: vocab/credentials/v2/vocabulary.yml
Comment (TallTed) [inline vocab/credentials/v2/vocabulary.yml:36]:
```suggestion
    comment: A type of validator for the validation of <a href="#VerifiableCredential">Verifiable Credentials</a> that are based on JSON Schema.
```

File: vocab/credentials/v2/vocabulary.yml
Comment (TallTed) [inline vocab/credentials/v2/vocabulary.yml:62]: With better understanding of the overall changes made by this PR, and noting that this term links back to its normative definition, I'm OK with dropping the `comment`.
```suggestion
```

File: vocab/credentials/v2/vocabulary.yml
Comment (TallTed) [reply vocab/credentials/v2/vocabulary.yml:133]: I wonder whether this was added during the expansion of the issuer-holder-verifier trinity to include coordinator, service, etc., for each?

**Review by TallTed — COMMENTED** [review]
Modulo the small changes added today, I'm OK with this being merged. I think it represents an overall improvement.

**Comment by msporny** [comment]
Normative, multiple reviews, changes requested and made, no objections, merging.

**Pull request merged**
